net-misc/tigervnc/files/tigervnc-1.12.0-xsession-path.patch

```diff
@@ -0,0 +1,14 @@
+Add more possible paths for XServer, depending on which display manager is
+installed
+
+--- a/unix/vncserver/vncserver.in
++++ b/unix/vncserver/vncserver.in
+@@ -437,7 +437,7 @@
+ 	die "$prog: couldn't find \"$cmd\" on your PATH.\n";
+     }
+ 
+-    foreach $cmd ("/etc/X11/xinit/Xsession", "/etc/X11/Xsession") {
++    foreach $cmd ("/etc/gdm/Xsession", "/etc/lightdm/Xsession", "/usr/share/sddm/scripts/Xsession", "/usr/share/slim/Xsession", "/etc/X11/Sessions/Xsession", "/etc/X11/xinit/Xsession", "/etc/X11/Xsession") {
```

Comment:
None on the original `Xsession` files exist on my system, I assume they are from other distributions.  `vncserver` needs an `Xsession` that will execute whatever parameters are passed to its command line.  Most do this, except the one that comes with `xinit`.  So I'll add here all `Xsession` scripts known to me.  There may be others.

Reply:
Automatically selecting the first valid Xsession from a list many cause issues for users having more than one display manager installed. We should either use /etc/conf.d/xdm to select the correct DM or, for preference, some new setting in /etc/conf.d/tigervnc

Reply:
I don't think it really matters that much.  All 4 Xsession files do essentially the same thing, the important sequence that makes `vncserver` start the correct session is this:
```
command="$@"
exec ${command}
```
And they all have it in some form.  The rest are only implementation details.

But yes, I can try to add an optional variable in `/etc/conf.d/tigervnc` to override the search.

Reply:
My suggestions in https://bugs.gentoo.org/746227#c41 to c49 do this and allow the person installing tigervnc to select different Xsession files for different system users. They can even create their own bespoke Xsession file.

Reply:
That looks a bit excessive, DISPLAY_MANAGER is a global setting.  Users with special needs can use their ~/.xsession file (or equivalent).
I've set sddm first and added a variable to override it.

Reply:
Sorry about that, I don't know what happened above. I'll try again.
I used /etc/conf.d/tigervnc to allow flexible overridde

\# Config file for /etc/init.d/tigervnc
\# Specify the user(s) Xvnc(1) should be run for. 
\# The syntax is: "username:display[,<Xsession-file>]"
\# example: "DISPLAYS="foo:1 bar:2,/usr/share/sddm/scripts/Xsession" You can specify more users separated by space.
\# DISPLAYS="myuser:1"
\# DISPLAYS="myuser:1,/usr/share/sddm/scripts/Xsession"